filebrowser/rockcraft.yaml

```diff
@@ -0,0 +1,158 @@
+  for more information.
+license: Apache-2.0
+run-user: _daemon_  # UID/GID 584792
+
+platforms:
+    amd64:
+
+services:
+  filebrowser:
+    override: merge
+    command: "/filebrowser [ ]"
+    working-dir: "/"
+    startup: enabled
+    on-check-failure:
+      alive: restart
+entrypoint-service: filebrowser
+
+checks:
+  alive:
+    override: replace
+    level: alive
+    period: 5s
+    exec:
+      command: /healthcheck.sh
+      working-dir: /
+      user: _daemon_
+      group: _daemon_
+
+parts:
+  security-team-requirement:
+    plugin: nil
+    override-build: |
+      mkdir -p ${CRAFT_PART_INSTALL}/usr/share/rocks
+      (echo "# os-release" && cat /etc/os-release && echo "# dpkg-query" && dpkg-query -f '${db:Status-Abbrev},${binary:Package},${Version},${source:Package},${Source:Version}\n' -W) > ${CRAFT_PART_INSTALL}/usr/share/rocks/dpkg.query
+
+  healthcheck:
+    plugin: nil
+    source: https://github.com/filebrowser/filebrowser.git
+    source-tag: v2.27.0
+    override-build: |
+      cp healthcheck.sh $CRAFT_PART_INSTALL/healthcheck.sh
+    stage-packages:
+      - curl
+      - jq
+    permissions:
+      - path: healthcheck.sh
+        # 584792 is the _daemon_ user
+        owner: 584792
+        group: 584792
+        mode: "755"
+
+  add-config:
+    plugin: nil
+    source: https://github.com/filebrowser/filebrowser.git
+    source-tag: v2.27.0
+    build-packages:
+      - jq
+    override-build: |
+      jq '.database = "/tmp/database.db"' docker_config.json > $CRAFT_PART_INSTALL/.filebrowser.json
+    # TODO: Omitting stage here means .filebrowser.json does not get copied automatically.  Not sure why
+    #       Asked here: https://matrix.to/#/!NPPCseDHKRvSBMUEXN:ubuntu.com/$3pkoT_gtSQlIa_J2lv6iy3P0JdzMQ5Cbt38guXG9Sg4?via=ubuntu.com&via=matrix.org
+    stage:
+      - ".filebrowser.json"
+
+  frontend:
+    # Tried to use `plugin: npm` but it does:
+    #  * downloads node
+    #  * `npm pack`
+    # `npm pack` only tars the dist, it does not actually build anything.  We
+    # need to run `npm run build`.  Instead, we use `plugin: nil` and download
+    # node manually.
+    # plugin: npm
+    # npm-include-node: true
+    # npm-node-version: "18.19.0"
+    plugin: nil
+    source: https://github.com/filebrowser/filebrowser.git
+    source-tag: v2.27.0
+    # tried to use source-subdir, but override-build ignores it.  Maybe source-subdir is
+    # implemented as part of the default build step?
+    # source-subdir: frontend
+    build-environment:
+      # Set the node version for build-override here.
+      - NODE_VERSION: '18.18.2'
+    override-build: |
+      cd frontend
+
+      # Setup node/npm
+      node_uri="https://nodejs.org/dist/v${NODE_VERSION}/node-v${NODE_VERSION}-linux-x64.tar.gz"
+      echo fetching node for build: $node_uri
+      # Download and unpack right here
+      curl $node_uri | tar xzf - -C ./ --no-same-owner --strip-components=1
+      PATH=$(pwd)/bin:$PATH
```

Comment:
Can we install npm as a `build-package`? I found this [example](https://github.com/canonical/traefik-rock/blob/9e9ded585dc16d1f27374b79f2d20e91d883dd4e/2.10.6/rockcraft.yaml#L31-L33) (or even [snap](https://github.com/canonical/prometheus-rock/blob/70c0ca78898df2849ffd7f039a19cf1b6dddcba9/2.46.0/rockcraft.yaml#L21-L23))

Reply:
Great question - hadn't spotted that!  Probably we don't care about the minor version so snap sounds helpful.  Let me try it

Reply:
Oh that's excellent, it worked great.  ty!  changed it to a snap (wasn't sure how to pin a version for the `build-packages`)